src/prommis/nanofiltration/diafiltration_two_salt.py

```diff
@@ -0,0 +1,870 @@
+                (
+                    self.config.feed_flow_volume * self.config.feed_conc_mass_lithium
+                    + self.config.diafiltrate_flow_volume
+                    * self.config.diafiltrate_conc_mass_lithium
+                )
+                / (self.config.feed_flow_volume + self.config.diafiltrate_flow_volume)
+            )
+
+        self.initial_retentate_conc_mass_lithium = Constraint(
+            rule=_initial_retentate_conc_mass_lithium
+        )
+
+        def _initial_retentate_conc_mass_cobalt(self):
+            return self.retentate_conc_mass_cobalt[0] == (
+                (
+                    self.config.feed_flow_volume * self.config.feed_conc_mass_cobalt
+                    + self.config.diafiltrate_flow_volume
+                    * self.config.diafiltrate_conc_mass_cobalt
+                )
+                / (self.config.feed_flow_volume + self.config.diafiltrate_flow_volume)
+            )
+
+        self.initial_retentate_conc_mass_cobalt = Constraint(
+            rule=_initial_retentate_conc_mass_cobalt
+        )
+
+        def _initial_membrane_interface_lithium(self):
+            return self.membrane_conc_mass_lithium[0, 0] == (0 * units.kg / units.m**3)
+
+        self.initial_membrane_interface_lithium = Constraint(
+            rule=_initial_membrane_interface_lithium
+        )
+
+        def _initial_membrane_interface_cobalt(self):
+            return self.membrane_conc_mass_cobalt[0, 0] == (0 * units.kg / units.m**3)
+
+        self.initial_membrane_interface_cobalt = Constraint(
+            rule=_initial_membrane_interface_cobalt
+        )
+
+        def _initial_membrane_interface_chlorine(self):
+            return self.membrane_conc_mass_chlorine[0, 0] == (0 * units.kg / units.m**3)
+
+        self.initial_membrane_interface_chlorine = Constraint(
+            rule=_initial_membrane_interface_chlorine
+        )
+
+        def _initial_permeate_conc_mass_lithium(self):
+            return self.permeate_conc_mass_lithium[0] == (0 * units.kg / units.m**3)
+
+        self.initial_permeate_conc_mass_lithium = Constraint(
+            rule=_initial_permeate_conc_mass_lithium
+        )
+
+        def _initial_permeate_conc_mass_cobalt(self):
+            return self.permeate_conc_mass_cobalt[0] == (0 * units.kg / units.m**3)
+
+        self.initial_permeate_conc_mass_cobalt = Constraint(
+            rule=_initial_permeate_conc_mass_cobalt
+        )
+
+        def _initial_d_retentate_conc_mass_lithium_dx(self):
+            return self.d_retentate_conc_mass_lithium_dx[0] == (
+                0 * units.kg / units.m**3
+            )
+
+        self.initial_d_retentate_conc_mass_lithium_dx = Constraint(
+            rule=_initial_d_retentate_conc_mass_lithium_dx
+        )
+
+        def _initial_d_retentate_conc_mass_cobalt_dx(self):
+            return self.d_retentate_conc_mass_cobalt_dx[0] == (
+                0 * units.kg / units.m**3
```

Comment:
As mentioned above, it's helpful to initialize with a very small non-zero value instead of zero.